.github/workflows/lint_and_test.yml

```
@@ -3,13 +3,7 @@
 
 name: lint and test
 
-on:
-  push:
-    branches:
-      - main
-  pull_request:
-    branches:
-      - main
+on: [push, pull_request]
```

```suggestion
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
  workflow_dispatch:
    inputs:
      branch:
        description: 'Branch to run on'
        required: true
        default: 'main'
```
Saw your comment about duplicating tests, that's not ideal.

What about instead allowing you to manually run the tests on a particular branch?

---

I found an alternative approach here: https://wildwolf.name/github-actions-how-to-avoid-running-the-same-workflow-multiple-times/